### DISCUSSION TITLE
Option to have DatabaseMigrator report if the database was erased due to a schema change

### DISCUSSION THREAD

**Discussion by Jnosh** [groue/GRDB.swift#1650] [Ideas]
`DatabaseMigrator` has the `eraseDatabaseOnSchemaChange` option to erase the database contents if the database schema changed.

I am using this option during development but I also have some non-DB data that I need to erase if this was exercised. Currently I am checking for the existence of a record in a single row table to detect if the DB contents were erased but it would be nicer if `DatabaseMigrator` could report this directly.

Do you think it would make sense to add this capability?

**Comment by groue** [comment]
Hello @Jnosh,

It looks like your app is using `eraseDatabaseOnSchemaChange` in release builds, on purpose. This is unfrequent, because apps are supposed to register new migrations instead of modifying existing ones, but maybe you do have a good reason for that.

Did you have a desired API in mind? May I also ask which action the app is performing when the database is erased (more context always help)?

**Reply by Jnosh** [reply]
I'm just using it during development (in DEBUG builds).

The auxiliary data is files stored by the application (which the DB tracks) and items indexed via Spotlight. If I change the DB schema during development and the DB is reset, I also want to erase these files and the Spotlight index so they are in sync with the DB contents.

**Reply by groue** [reply]
OK. Do you think #1651 could address your needs?

**Reply by Jnosh** [reply]
Yes! That's even more flexible than what I had in mind. Thank you very much!

**Reply by groue** [reply]
Thank you for the feature request! The proposed api allows you to handle db values before and after the call to `erase`, and I'm glad it fits your needs :) It will be shipped shortly!

**Reply by groue** [reply]
Shipped in [v7.0.0-beta.4](https://github.com/groue/GRDB.swift/releases/tag/v7.0.0-beta.4)!
